x-pack/plugins/security_solution/public/timelines/components/timeline/header/title_and_description.tsx

```diff
@@ -167,57 +204,53 @@ export const TimelineTitleAndDescription = React.memo<TimelineTitleAndDescriptio
             )}
-            <EuiFlexItem grow={true}>
-              <EuiFormRow label={TIMELINE_TITLE}>
-                <Name
-                  autoFocus={initialFocus === 'title'}
-                  disableTooltip={true}
-                  disableAutoSave={true}
-                  disabled={isSaving}
-                  data-test-subj="save-timeline-name"
-                  timelineId={timelineId}
+            <Form form={form}>
+              <EuiFlexItem grow={true}>
+                <CommonUseField
+                  path="title"
+                  fullWidth
+                  label={i18n.TITLE}
+                  euiFieldProps={titleFieldProps}
                 />
-              </EuiFormRow>
-              <EuiSpacer />
-            </EuiFlexItem>
-            <EuiFlexItem grow={true}>
-              <EuiFormRow label={descriptionLabel}>
-                <Description
-                  autoFocus={initialFocus === 'description'}
-                  data-test-subj="save-timeline-description"
-                  disableTooltip={true}
-                  disableAutoSave={true}
-                  disabled={isSaving}
-                  timelineId={timelineId}
+                <EuiSpacer />
+              </EuiFlexItem>
+              <EuiFlexItem grow={true}>
+                <CommonUseField
+                  label={descriptionLabel}
+                  path="description"
+                  fullWidth
+                  euiFieldProps={descriptionFieldProps}
                 />
-              </EuiFormRow>
-              <EuiSpacer />
-            </EuiFlexItem>
-            <EuiFlexItem grow={false}>
-              <EuiFlexGroup justifyContent="flexEnd">
-                <EuiFlexItem grow={false} component="span">
-                  <EuiButton
-                    fill={false}
-                    onClick={handleCancel}
-                    isDisabled={isSaving}
-                    data-test-subj="close-button"
-                  >
-                    {closeModalText}
-                  </EuiButton>
-                </EuiFlexItem>
-                <EuiFlexItem grow={false} component="span">
-                  <EuiButton
-                    isDisabled={title.trim().length === 0 || isSaving}
-                    fill={true}
-                    onClick={handleClick}
-                    data-test-subj="save-button"
-                  >
-                    {saveButtonTitle}
-                  </EuiButton>
-                </EuiFlexItem>
-              </EuiFlexGroup>
-            </EuiFlexItem>
-          </Wrapper>
+                <EuiSpacer />
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiFlexGroup justifyContent="flexEnd" gutterSize="s">
+                  <EuiFlexItem grow={false} component="span">
+                    <EuiButton
+                      size="s"
+                      fill={false}
+                      onClick={handleCancel}
+                      isDisabled={isSaving}
```

Comment:
nit -> i think I will also disable it when isSubmitting is true

Reply:
because our actions are not async, `isSubmitting` will resolve immediately